roles/kubernetes-apps/ansible/tasks/cleanup_dns.yml

```diff
@@ -1,4 +1,12 @@
 ---
+- name: Kubernetes Apps | Register coredns deployment annotation `createdby`
+  shell: "{{ bin_dir }}/kubectl get deploy -n kube-system coredns -o jsonpath='{ .spec.template.metadata.annotations.createdby }'"
+  register: createdby_annotation
+  delegate_to: "{{ groups['kube-master']|first }}"
```

Comment:
we have a problem )))
classic error with ansible delegate_to task

is this task you register var `createdby_annotation` for random host.

but in next var you check var, defined for host 'kube-master'[0]

please remove delegate_to and run_once, 
and add
```
  when:
    - dns_mode in ['coredns', 'coredns_dual']
    - inventory_hostname == groups['kube-master'][0]
```

Reply:
If I print out the `createdby_annotation` var on every host in the next task, each host has the correct value for the variable. Are you expecting only a single node would know this var? 

I'll get your requested changes in, just want to make sure I understand this ansible issue for the future.